### PR TITLE
release-dev

### PR DESCRIPTION
**Pull request by biaMurNtt**

### CODE REVIEW

File: connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java [hunk 1]
```diff
@@ -38,34 +61,55 @@ class AzureApiManagerClientTest {
...
+                .thenReturn(subscriptionContractListMock);
+
+//        when(managerMock.userSubscriptions().list(any(), any(), any()).stream().map())
+//                .thenReturn(subscriptionContractListMock);
+
         when(managerMock.subscriptions().listSecretsWithResponse(any(), any(), any(), any()))
                 .thenReturn(keyContract);
+//        when(apiManagerClient.createInstitutionSubscription(any(), any()))
+//                .thenReturn(apiKeysMockList);
+
```
Comment (KevinSi96) [inline connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java:103]: remove comments

File: connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java [hunk 2]
```diff
@@ -74,10 +118,77 @@ void createInstitutionSubscription_ok() throws NoSuchFieldException, IllegalAcce
...
+        SubscriptionsImpl subscriptionMock = mock(SubscriptionsImpl.class);////////////////////////////
+        UserSubscriptionsImpl userSubscriptionsMock = mock(UserSubscriptionsImpl.class);
+        AzureApiManagerClient managerClient = new AzureApiManagerClient(serviceName,
+                resourceGroup,
+                subscriptionId,
+                tenantId);
+        SubscriptionContract contractMock = mock(SubscriptionContract.class);
+        //DummyKeyContract keyContract = mockInstance(new DummyKeyContract());
+        // PagedIterable<SubscriptionContract> subscriptionContractList = getPi(1);
```
Comment (KevinSi96) [inline connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java:138]: remove comments

File: connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java [hunk 3]
```diff
@@ -74,10 +118,77 @@ void createInstitutionSubscription_ok() throws NoSuchFieldException, IllegalAcce
...
+        when(managerMock.userSubscriptions().list(any(), any(), any()))
+                .thenReturn(subscriptionContractListMock);
+
+//        when(managerMock.userSubscriptions().list(any(), any(), any()).stream().map())
+//                .thenReturn(subscriptionContractListMock);
+
+//        when(managerMock.subscriptions().listSecretsWithResponse(any(), any(), any(), any()))
+//                .thenReturn(keyContract);
+//        when(apiManagerClient.createInstitutionSubscription(any(), any()))
+//                .thenReturn(apiKeysMockList);
```
Comment (KevinSi96) [inline connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java:168]: remove comments

File: connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java [hunk 4]
```diff
@@ -74,10 +118,77 @@ void createInstitutionSubscription_ok() throws NoSuchFieldException, IllegalAcce
...
+
+        //when
+        managerClient.createInstitutionSubscription(institutionId, institutionName, scope, subscriptionId, serviceName);
+        //then
+//        assertNotNull(apiKeys);
+//        assertNotNull(apiKeys.get(0).getPrimaryKey());
+//        assertNotNull(apiKeys.get(0).getSecondaryKey());
+//        assertEquals(keyContract.getValue().primaryKey(), apiKeys.get(0).getPrimaryKey());
+//        assertEquals(keyContract.getValue().secondaryKey(), apiKeys.get(0).getSecondaryKey());
```
Comment (KevinSi96) [inline connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java:177]: why are these commented?

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/ApiManagerMapper.java [hunk 5]
```diff
@@ -11,8 +14,14 @@ public static ApiKeysResource toApiKeysResource(InstitutionApiKeys model) {
             resource = new ApiKeysResource();
             resource.setPrimaryKey(model.getPrimaryKey());
             resource.setSecondaryKey(model.getSecondaryKey());
+            resource.setDisplayName(model.getDisplayName());
+            resource.setId(model.getId());
```
Comment (KevinSi96) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/model/mapper/ApiManagerMapper.java:18]: Add test for these two fields in the mapper

File: web/src/test/java/it/pagopa/selfcare/pagopa/backoffice/web/model/subscriptions/ApiKeysResourceTest.java [hunk 6]
```diff
@@ -35,9 +35,11 @@ void validateNullFields(){
         HashMap<String, Class<? extends Annotation>> toCheckMap = new HashMap<>();
         toCheckMap.put("primaryKey", NotBlank.class);
         toCheckMap.put("secondaryKey", NotBlank.class);
+        toCheckMap.put("displayName", NotBlank.class);
```
Comment (KevinSi96) [inline web/src/test/java/it/pagopa/selfcare/pagopa/backoffice/web/model/subscriptions/ApiKeysResourceTest.java:38]: missing check on id

**Review by KevinSi96 — CHANGES_REQUESTED** [review]

File: connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java (hunk 4 above)
Comment (biaMurNtt) [reply connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java:177]: cause the method is void and there are no fields to check

File: connector/azure-apimanager/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClient.java [hunk 7]
```diff
@@ -87,21 +90,38 @@ public InstitutionApiKeys createInstitutionSubscription(String institutionId, St
...
+    @Override
+    public List<InstitutionApiKeys> getInstitutionApiKeys(String institutionId) {
         log.trace("getInstitutionApiKeys start");
         log.debug("getInstitutionApiKeys serviceName = {}, resourceGroup = {}, institutionId = {}", serviceName, resourceGroupName, institutionId);
-        InstitutionApiKeys subscription = getApiKeys(institutionId);
-        log.debug(LogUtils.CONFIDENTIAL_MARKER,"getUser result = {}", subscription);
-        return subscription;
+        List<InstitutionApiKeys> subscriptions = getApiSubscriptions(institutionId);
+        log.debug(LogUtils.CONFIDENTIAL_MARKER, "getUser result = {}", subscriptions);
+        return subscriptions;
```
Comment (KevinSi96) [inline connector/azure-apimanager/src/main/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClient.java:121]: correct the method name and add log.trace to the end

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java [hunk 8]
```diff
@@ -85,8 +85,8 @@ public void regeneratePrimaryKey(@ApiParam("${swagger.model.institution.id}")
     @PostMapping("/{institutionId}/api-keys/secondary/regenerate")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @ApiOperation(value = "", notes = "${swagger.api.institution.regenerateSecondaryKey}")
-    public void regenerateSecondaryKey(@ApiParam("${swagger.model.institution.id}")
-                                     @PathVariable("institutionId")String institutionId){
+    public void regenerateSecondaryKey(@ApiParam("${swagger.model.institution.subscription.id}")
+                                       @PathVariable("institutionId") String institutionId) {
```
Comment (KevinSi96) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java:89]: swagger description changed to subscription Id but path variable stayed institutionId

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java [hunk 9]
```diff
@@ -60,22 +60,22 @@ public ApiKeysResource getInstitutionApiKeys(
...
+        return apiKeysResourceList;
     }
     
     @PostMapping("/{institutionId}/api-keys/primary/regenerate")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @ApiOperation(value = "", notes = "${swagger.api.institution.regeneratePrimaryKey}")
-    public void regeneratePrimaryKey(@ApiParam("${swagger.model.institution.id}")
-                                     @PathVariable("institutionId")String institutionId){
+    public void regeneratePrimaryKey(@ApiParam("${swagger.model.institution.subscription.id}")
+                                     @PathVariable("institutionId") String institutionId) {
```
Comment (KevinSi96) [inline web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java:78]: swagger description changed to subscriptionId but path variable stayed institutionId why?

**Review by KevinSi96 — CHANGES_REQUESTED** [review]

File: web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java (hunk 9 above)
Comment (biaMurNtt) [reply web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/InstitutionController.java:78]: fixed

**Review by KevinSi96 — APPROVED** [review]

**Pull request merged**